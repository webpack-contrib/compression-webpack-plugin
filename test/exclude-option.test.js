import CompressionPlugin from '../src/index';

import {
  compile,
  getAssetsNameAndSize,
  getCompiler,
  getErrors,
  getWarnings,
} from './helpers/index';

describe('"exclude" option', () => {
  let compiler;

  beforeEach(() => {
    compiler = getCompiler(
      './entry.js',
      {},
      {
        output: {
          path: `${__dirname}/dist`,
          filename: '[name].js?var=[hash]',
          chunkFilename: '[id].[name].js?ver=[hash]',
        },
      }
    );
  });

  it('matches snapshot for a single `exclude` value ({RegExp})', async () => {
    new CompressionPlugin({
      exclude: /\.svg(\?.*)?$/i,
      minRatio: 1,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });

  it('matches snapshot for multiple `exclude` values ({Array<RegExp>})', async () => {
    new CompressionPlugin({
      exclude: [/\.svg(\?.*)?$/i, /\.png(\?.*)?$/i],
      minRatio: 1,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });
});
