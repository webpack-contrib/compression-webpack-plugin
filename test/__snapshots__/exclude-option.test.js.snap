// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`exclude\` option matches snapshot for a single \`exclude\` value ({RegExp}): assets 1`] = `
Array [
  Array [
    "0.async.js.gz?ver=81982f432266ef1edd1a",
    117,
  ],
  Array [
    "0.async.js?ver=81982f432266ef1edd1a",
    130,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    73160,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "js.js.gz?var=81982f432266ef1edd1a",
    2802,
  ],
  Array [
    "js.js?var=81982f432266ef1edd1a",
    10221,
  ],
]
`;

exports[`when applied with \`exclude\` option matches snapshot for a single \`exclude\` value ({RegExp}): errors 1`] = `Array []`;

exports[`when applied with \`exclude\` option matches snapshot for a single \`exclude\` value ({RegExp}): warnings 1`] = `Array []`;

exports[`when applied with \`exclude\` option matches snapshot for multiple \`exclude\` values ({Array<RegExp>}): assets 1`] = `
Array [
  Array [
    "0.async.js.gz?ver=81982f432266ef1edd1a",
    117,
  ],
  Array [
    "0.async.js?ver=81982f432266ef1edd1a",
    130,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "js.js.gz?var=81982f432266ef1edd1a",
    2802,
  ],
  Array [
    "js.js?var=81982f432266ef1edd1a",
    10221,
  ],
]
`;

exports[`when applied with \`exclude\` option matches snapshot for multiple \`exclude\` values ({Array<RegExp>}): errors 1`] = `Array []`;

exports[`when applied with \`exclude\` option matches snapshot for multiple \`exclude\` values ({Array<RegExp>}): warnings 1`] = `Array []`;
